cmd/juju/crossmodel/find.go

```diff
@@ -191,7 +193,7 @@ type ApplicationOfferResult struct {
 
 func accessForUser(user names.UserTag, users []crossmodel.OfferUserDetails) string {
 	for _, u := range users {
-		if u.UserName == user.Id() {
+		if strings.ToLower(u.UserName) == strings.ToLower(user.Id()) {
```

Comment:
I'm not sure this is the correct fix for this. I'd want to know why `user.Id` is different from the crossmodel user details.

Reply:
When juju accepts a new user, it lower cases the user name to get the user id. The API call to get the offer details returns the Id() in the UserName field from what I can see - I think the correct fix would be to look at fixing the facade implementation.

Reply:
it's just that external identities are not always all lower case as juju assumes..

Reply:
That's what @wallyworld is saying, we should fix it where we ingest the user as this might not be the only location where we're getting `u.UserName` and `user.Id()` lookup wrong.

Reply:
agreed. this is probably not the only place. please close this PR when a proper fix is proposed.